From patchwork Wed Sep  5 23:28:00 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liran Alon <liran.alon@oracle.com>
X-Patchwork-Id: 10589623
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7F3AC15E9
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  5 Sep 2018 23:28:38 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 61B672ABF3
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  5 Sep 2018 23:28:38 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 557C62AC00; Wed,  5 Sep 2018 23:28:38 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	UNPARSEABLE_RELAY autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id E4FE32ABF3
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed,  5 Sep 2018 23:28:37 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1725804AbeIFEBE (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 6 Sep 2018 00:01:04 -0400
Received: from userp2120.oracle.com ([156.151.31.85]:51994 "EHLO
        userp2120.oracle.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725741AbeIFEBD (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 6 Sep 2018 00:01:03 -0400
Received: from pps.filterd (userp2120.oracle.com [127.0.0.1])
        by userp2120.oracle.com (8.16.0.22/8.16.0.22) with SMTP id
 w85NJ5Pr067514;
        Wed, 5 Sep 2018 23:28:33 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id; s=corp-2018-07-02;
 bh=TRPgecJJomB+ARLdycD0JZDQGFa2xNx+XjSqJP8mt+o=;
 b=Fbpx++tk2KAj8uUXUkCZluhSL5wYBH0Q70LyvoDoSJUuUiiz+n1a8Uf7iemIA/KFI2mu
 vVm/ULkNy1paG7aa4AGpz9CmVTiR2jU5x2FrXb5YTZqJeVrX1PTL4mvXX5PVUj0KFOLh
 MxMSStjlhQvTMjQ4+I7hEa5hy3XBZev6dLLtMS1MeqWSv69wEuiDUuw1ATrILVhitpxO
 vxeLw/Aif+1wzwAk9tOsGQv030XUEFyJjTtZrt8QpJHt1xRmK0acN8Mx8IAHtIzKxQjE
 fcwbCZXRLmFYr99wHjUaKzTSu6e29VGE8XkDBd/h9t/XOpfZYBRX13ncComorCYwai6s ow==
Received: from userv0021.oracle.com (userv0021.oracle.com [156.151.31.71])
        by userp2120.oracle.com with ESMTP id 2m7kdqprt1-1
        (version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 05 Sep 2018 23:28:33 +0000
Received: from userv0121.oracle.com (userv0121.oracle.com [156.151.31.72])
        by userv0021.oracle.com (8.14.4/8.14.4) with ESMTP id w85NSXoa011015
        (version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-GCM-SHA384 bits=256
 verify=OK);
        Wed, 5 Sep 2018 23:28:33 GMT
Received: from abhmp0012.oracle.com (abhmp0012.oracle.com [141.146.116.18])
        by userv0121.oracle.com (8.14.4/8.13.8) with ESMTP id w85NSWId007131;
        Wed, 5 Sep 2018 23:28:32 GMT
Received: from spark.ravello.local (/213.57.127.2)
        by default (Oracle Beehive Gateway v4.0)
        with ESMTP ; Wed, 05 Sep 2018 16:28:32 -0700
From: Liran Alon <liran.alon@oracle.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com, kvm@vger.kernel.org
Cc: idan.brown@oracle.com, Liran Alon <liran.alon@oracle.com>
Subject: [kvm-unit-tests PATCH] x86: UMIP: Fix wrong print of CR4.UMIP bit
 status
Date: Thu,  6 Sep 2018 02:28:00 +0300
Message-Id: <20180905232800.10318-1-liran.alon@oracle.com>
X-Mailer: git-send-email 2.16.1
X-Proofpoint-Virus-Version: vendor=nai engine=5900 definitions=9007
 signatures=668708
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 suspectscore=0
 malwarescore=0
 phishscore=0 bulkscore=0 spamscore=0 mlxscore=0 mlxlogscore=753
 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.0.1-1807170000 definitions=main-1809050221
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Reviewed-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
Signed-off-by: Liran Alon <liran.alon@oracle.com>
Reviewed-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 x86/umip.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/x86/umip.c b/x86/umip.c
index cda19714579f..357ca3cbe065 100644
--- a/x86/umip.c
+++ b/x86/umip.c
@@ -187,8 +187,8 @@ int main()
     }
     write_cr4(read_cr4() | X86_CR4_UMIP);
 
-    test_umip_nogp("UMIP=0, CPL=0\n");
-    do_ring3(test_umip_gp, "UMIP=0, CPL=3\n");
+    test_umip_nogp("UMIP=1, CPL=0\n");
+    do_ring3(test_umip_gp, "UMIP=1, CPL=3\n");
 
     return report_summary();
 }
